### PR TITLE
extend logs for ReadReportForCluster

### PR DESCRIPTION
**Pull request by Bee-lee**
# Description
still debugging prod issue for 404s. The error happens in the ReadReportForCluster function

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

## Testing steps
`make run`

## Checklist
* [ ] `make before_commit` passes
* [ ] updated documentation wherever necessary
* [ ] added or modified tests if necessary
* [ ] updated schemas and validators in [insights-data-schemas](https://github.com/RedHatInsights/insights-data-schemas) in case of input/output change

### CODE REVIEW

**Review by dpensi — APPROVED** [review]

File: server/server.go [hunk 1]
```diff
@@ -160,7 +161,7 @@ func (server *HTTPServer) readReportForCluster(writer http.ResponseWriter, reque
 		clusterName,
 	)
 	if err != nil {
-		log.Error().Err(err).Msg("Unable to read report for cluster")
+		log.Error().Err(err).Msgf("Unable to read report for cluster %v", clusterName)
```
Comment (juandspy) [inline server/server.go:164]: prob. it would be better to use `.Str("cluster", clusterName)`

File: server/server.go [hunk 2]
```diff
@@ -144,6 +144,7 @@ func (server *HTTPServer) readReportForCluster(writer http.ResponseWriter, reque
 		// everything has been handled already
 		return
 	}
+	log.Info().Msgf("readReportForCluster start for cluster %v", clusterName)
```
Comment (juandspy) [inline server/server.go:147]: prob. it would be better to use .Str("cluster", clusterName)

File: storage/storage.go [hunk 3]
```diff
@@ -531,6 +531,8 @@ func parseRuleRows(rows *sql.Rows) ([]types.RuleOnReport, error) {
 		report = append(report, rule)
 	}
 
+	log.Info().Msgf("parseRuleRows final report %v", report)
```
Comment (juandspy) [inline storage/storage.go:534]: prob. it would be better to use .Str("report", report)

File: storage/storage.go [hunk 4]
```diff
@@ -680,6 +682,8 @@ func (storage DBStorage) ReadReportsForClusters(clusterNames []types.ClusterName
 func (storage DBStorage) ReadReportForCluster(
 	orgID types.OrgID, clusterName types.ClusterName,
 ) ([]types.RuleOnReport, types.Timestamp, types.Timestamp, types.Timestamp, error) {
+	log.Info().Msgf("ReadReportForCluster start for cluster %v", clusterName)
```
Comment (juandspy) [inline storage/storage.go:685]: prob. it would be better to use `.Str("cluster", clusterName)`

File: storage/storage.go [hunk 5]
```diff
@@ -690,6 +694,7 @@ func (storage DBStorage) ReadReportForCluster(
 		"SELECT last_checked_at, reported_at, gathered_at FROM report WHERE org_id = $1 AND cluster = $2;",
 		orgID, clusterName,
 	).Scan(&lastChecked, &reportedAt, &gatheredAtInDB)
+	log.Info().Msgf("ReadReportForCluster query from report table raw DB error %v for cluster %v", err, clusterName)
```
Comment (juandspy) [inline storage/storage.go:697]: prob. it would be better to use `.Str("cluster", clusterName)` and `.Err(err)`

File: storage/storage.go [hunk 6]
```diff
@@ -704,15 +709,18 @@ func (storage DBStorage) ReadReportForCluster(
...
 	}
 
 	rows, err := storage.connection.Query(
 		"SELECT template_data, rule_fqdn, error_key, created_at FROM rule_hit WHERE org_id = $1 AND cluster_id = $2;", orgID, clusterName,
 	)
+	log.Info().Msgf("ReadReportForCluster query from rule_hit table raw DB error %v for cluster %v", err, clusterName)
 
 	err = types.ConvertDBError(err, []interface{}{orgID, clusterName})
 	if err != nil {
+		log.Info().Msgf("ReadReportForCluster query from rule_hit table converted DB error %v for cluster %v", err, clusterName)
```
Comment (juandspy) [inline storage/storage.go:723]: prob. it would be better to use `.Str("cluster", clusterName)` and `.Err(err)`

File: storage/storage.go [hunk 7]
```diff
@@ -704,15 +709,18 @@ func (storage DBStorage) ReadReportForCluster(
...
 	err = types.ConvertDBError(err, []interface{}{orgID, clusterName})
 	if err != nil {
+		log.Error().Err(err).Msgf("ReadReportForCluster query from report table converted DB error %v for cluster %v", err, clusterName)
 		return report, lastCheckedStr, reportedAtStr, gatheredAtStr, err
 	}
 
 	rows, err := storage.connection.Query(
 		"SELECT template_data, rule_fqdn, error_key, created_at FROM rule_hit WHERE org_id = $1 AND cluster_id = $2;", orgID, clusterName,
 	)
+	log.Info().Msgf("ReadReportForCluster query from rule_hit table raw DB error %v for cluster %v", err, clusterName)
```
Comment (juandspy) [inline storage/storage.go:719]: prob. it would be better to use `.Str("cluster", clusterName)` and `.Err(err)`

File: storage/storage.go [hunk 8]
```diff
@@ -704,15 +709,18 @@ func (storage DBStorage) ReadReportForCluster(
 
 	err = types.ConvertDBError(err, []interface{}{orgID, clusterName})
 	if err != nil {
+		log.Error().Err(err).Msgf("ReadReportForCluster query from report table converted DB error %v for cluster %v", err, clusterName)
```
Comment (juandspy) [inline storage/storage.go:712]: prob. it would be better to use `.Str("cluster", clusterName)` and `.Err(err)`

**Review by joselsegura — APPROVED** [review]
Agree with Juan's comments, but still LGTM given the situation and time

**Review by tisnik — APPROVED** [review]
LGTM

**Pull request merged**